modules/pam_userdb/pam_userdb.c

```diff
@@ -213,30 +213,39 @@ user_lookup (pam_handle_t *pamh, const char *database, const char *cryptmode,
 	  } else if (ctrl & PAM_ICASE_ARG) {
 	    compare = -2;
 	  } else {
+		char *setting = malloc(data.dsize+1);
+		if (setting != NULL) {
+			strncpy(setting, data.dptr, data.dsize);
+			setting[data.dsize] = '\0';
+			if (ctrl & PAM_DEBUG_ARG) {
+				pam_syslog(pamh, LOG_INFO, "crypt setting: %s", setting);
+			}
```

Comment:
There are multiple problems here:
1. use strndup instead of malloc+strncpy
2. do not call it setting, that is misleading, call it salt or something
3. no need to syslog it as there is syslog of it above already
4. please correct the indentation so the whole block is aligned

Reply:
Ad 2.: I used it according to crypt_r documentation in man page and source code. I wasn't sure because `salt` might be misleading for some hashes, but I can rename it, you surely know way more about it.
Ad 4.: The whole file is a mess with respect to indentation, a lot of mixing of tabs and spaces. I wanted to make the change as minimal as possible so it is easy to review, basically using the same (mis-)indentation as in the lines around. I will be happy to fix it if you provide me some guidance - Should I do it withing this PR in separate commit or create a new one? Should I use tab or spaces (and how much?) - I wasn't able to find any guildeline. Should I fix it in whole file, just in the function, or just the respective block?

Reply:
`setting` might be correct if you're producing a new password hash. Here it is definitely not just the setting, it is the whole password hash from the DB. Use `pwhash` or something.

For 4. I mean just adding an indentation level to the whole block if you're adding an `if`. Not just indenting properly the new code but leaving the existing code that is inside this new block unindented as you did. Of course keep the rest of the code as is.